src/java.base/share/classes/java/lang/reflect/code/bytecode/SlotSSA.java

```diff
@@ -291,6 +294,20 @@ public static void findJoinPoints(Body body, Map<Block, Set<Integer>> joinPoints
         }
     }
 
+    /**
+     * Removes all catch blocks from join points to avoid adding block parameters to them.
+     *
+     * @param body the body
+     * @param joinPoints the join points to clean.
+     */
+    public static void removeCatchBlocksFromJoinPoints(Body body, Map<Block, Set<Integer>> joinPoints) {
```

Comment:
An alternative is to remove them from the dominance frontier map?

Reply:
The catch blocks are now removed from the dominance frontiers.